test/fail_compilation/impconv.d

```diff
@@ -0,0 +1,18 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/impconv.d(16): Error: function `impconv.foo(float)` is not callable using argument types `(int)`
+fail_compilation/impconv.d(17): Error: function `impconv.bar(double)` is not callable using argument types `(long)`
+---
+*/
+
+void foo(float);
+void bar(double);
```

Comment:
Add test for `real`?

Reply:
6 tests are suggested here: [float, double, real] * [int.max, long.max]

Reply:
I went for [1, int.max] for `float`, as long.max would just overflow as well as int.max already.

Wrt. `real`, that's compiler-specific as it depends on the front-end's `real_t` precision. Meaning that LDC and GDC would need to adapt that test, something I tried to avoid.

Reply:
The precision of `real_t` should be the same as the target platform (just not necessarily the same precision as the host). Having version ARM, MIPS, etc is safe.

Reply:
Not true for LDC in all cases. What happened to your huge common `real_t` type with ~160 bit mantissa?

Reply:
`real_t` is used for floats, doubles, and reals alike.  It may hold more precision, but when read, it's still rounded to match at the same precision as the target.